sympy/matrices/common.py

```diff
@@ -2248,12 +2262,13 @@ def __pow__(self, exp):
             # When certain conditions are met,
             # Jordan block algorithm is faster than
             # computation by recursion.
-            elif a.rows == 2 and exp > 100000 and jordan_pow is not None:
+            elif jordan_pow is not None and (jordan or \
+                    (jordan is not False and a.rows == 2 and exp > 100000)):
```

Comment:
We don't normally use `\` for line continuation in SymPy. It isn't needed when there are brackets (as above).